app/src/main/java/fr/free/nrw/commons/upload/UploadService.java

```diff
@@ -278,10 +278,11 @@ private void uploadContribution(Contribution contribution) {
                         showFailedNotification(contribution);
                     } else {
                         String canonicalFilename = "File:" + uploadResult.getFilename();
+                        String wikiItemName = contribution.getWikiItemName()!=null?contribution.getWikiItemName():"";
```

Comment:
I don't think you need to null check this, or at least not here.

`createClaimWithLogging` already null checks the `p18` and `wikidatEntityId`, all fields that are only present when the contribution has an associated `place` ie was uploaded from Nearby or we detected it was Nearby Place from the associated image data.

Reply:
It would be much nicer if we expressed `Contribution` & `ContributionWithAssociatedWikidataItem` as types (preferably better named ones too) instead of the current scenario. Far out of scope of this ticket, just me musing

Reply:
Sorry, Test cases 😅 
have to edit test file to pass values.

Reply:
@macgills  Done :-)

Reply:
Good work